tensorzero-optimizers/src/gepa/pareto.rs

```diff
@@ -0,0 +1,1760 @@
+//! Pareto frontier analysis and dominance checking for GEPA
+//!
+//! This module implements the core Pareto optimality logic for multi-objective optimization:
+//! - Instance-wise Pareto frontier computation
+//! - Global Pareto dominance checking
+//! - Missing data imputation
+
+use std::collections::{HashMap, HashSet};
+
+use evaluations::EvaluatorStats;
+use tensorzero_core::{
+    config::MetricConfigOptimize,
+    error::{Error, ErrorDetails},
+    evaluations::{EvaluatorConfig, InferenceEvaluationConfig},
+    variant::chat_completion::UninitializedChatCompletionConfig,
+};
+
+// Type aliases for cleaner score map signatures (TODO: will live next to evaluation code)
+pub type EvaluatorName = String;
+
+/// Unique identifier for a datapoint/example in a dataset
+pub type DatapointId = String;
```

Comment:
we should be doing more of this in the codebase honestly. One question though is why is DatapointId a string not a Uuid?